### PR TITLE
Upgrade typescript 4.0

### PR DESCRIPTION
**Pull request by johnschoeman**
Why:
We would like to be using the current versions of our developer tooling
as a matter of best practice.

This commit:
Upgrades typescript to 4.0.3 and updates all of the related tooling for
eslint-typescript which caused a need to update various globals
defined in the types for react native and react.

Co-Authored-By: devin jameson <devin@thoughtbot.com>

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,111 +1,116 @@
...
-      files: ['*.ts', '*.tsx'],
+      files: ["*.ts", "*.tsx"],
       rules: {
-        '@typescript-eslint/explicit-module-boundary-types': ['error'],
+        "@typescript-eslint/explicit-module-boundary-types": ["error"],
       },
     },
   ],
-};
+  globals: {
```
Comment (johnschoeman) [inline .eslintrc.js:111]: This is the only added config in this file.

**Review by mattThousand — APPROVED** [review]
👏 🥇

File: src/SelfAssessment/index.js [hunk 2]
```diff
@@ -76,6 +76,7 @@ const Assessment = ({ navigation }) => {
     // to support dynamic questions
 
     // memoize assessment question
+    /* eslint-disable react/display-name */
```
Comment (aledustet) [inline src/SelfAssessment/index.js]: I believe this comment could be avoided if we do not use an anonymous function, perhaps doing:
```
function assesmentQuestionMemoized({ navigation, route }) {
  return {...}
}
```
Or extracting the function into a named component might fix this?

File: src/components/GlobalText.tsx [hunk 3]
```diff
@@ -7,15 +7,15 @@ interface GlobalTextProps {
   style?: TextStyle
   testID?: string
   onPress?: () => void
-  children: JSX.Element | string
+  children: React.ReactNode | string
```
Comment (aledustet) [inline src/components/GlobalText.tsx]: very low priority but how do we feel about importing node from `React` as we are with the `FunctionComponent` ?

**Review by aledustet — APPROVED** [review]
👏👏 👏 👏  🔥🔥🔥🔥🔥

File: src/SelfAssessment/index.js (hunk 2 above)
Comment (johnschoeman) [reply src/SelfAssessment/index.js]: this file go removed anyways, so no longer an issue!

**Pull request merged**